Agents/FeatureInfoAgent/code/src/main/java/com/cmclinnovations/featureinfo/kg/TimeHandler.java

```diff
@@ -358,18 +362,25 @@ protected TimeSeries<Instant> buildTimeseriesObject(String measureIRI, Connectio
         TimeSeries<Instant> result = null;
         if(this.hours < 0) {
             // Get all data
+            LOGGER.debug("Getting timeseries without limit...");
             result = this.tsClient.getTimeSeries(new ArrayList<>(Arrays.asList(fixedIRI)), rdbConnection);
+
         } else {
+            LOGGER.debug("Getting timeseries with limit of last {} hours...", this.hours);
+
             // Determine bounds
             Instant lowerBound = LocalDateTime.now().minusHours(this.hours).toInstant(ZoneOffset.UTC);
             Instant upperBound = LocalDateTime.now().toInstant(ZoneOffset.UTC);
```

Comment:
Are we sure that this is what is required? Should it be something like taking the latest entry in the time series table and subtracting that instead?

Reply:
I'm inclined to agree with @kokfoongcmcl here. It is likely more useful to see the so many _most recent_ hours, rather than so many hours back from _now_ (which is likely to be nothing at all very quickly!).

Reply:
The current design we've used in all previous visualisations is to show data from the last 24 hours.

We can change this to range from N hours since the latest data point, but this is something I would require to be done within the TimeseriesClient (I suggest as a new method). Unless some derivation calculation is being made, I don't think agents should handle filtering the timeseries at all, this should be done centrally.

Reply:
The easiest thing to do now is to use getLatestData, get the value of the time from the TimeSeries object, then calculate the range and use getTimeSeriesWithinBounds.

It might be tricky to implement this in the TimeSeriesClient itself, because I allowed the Time class to be anything (e.g. Integer, Double, etc.), if it's Double, and you provide a Duration object, I don't know how to deal with it

Reply:
Would this trigger 2 separate connections to the RDB?

I strongly think the TimeseriesClient should be responsible for any filtering, as I don't want to tie any agents directly to the format in which timeseries are returned. Additionally, this sounds like something we'll want to use in different agents, so should be done centrally.

Reply:
You can use the same connection object for both calls.

It's not straightforward to support that, it will need to have one method for each class, I can't just do "-" unless they're subclass of Number

Reply:
I suggest this request goes on the back burner then until such time that it can be added centrally.

Reply:
As Seb said, we don't think this is a good idea anyway because you'll get empty results quite quickly, depending on the data of course